hw/syn/tools/dc/parse-syn-report.py

```diff
@@ -383,33 +434,28 @@ def get_results(args):
 
     results["top"] = args.dut
 
-    # flow messages
-    results = _parse_file(args.logpath, 'synthesis.log', results, _extract_messages,
-                          "flow", args)
-
-    # messages
-    for rep_type in ["analyze", "elab", "compile"]:
-        results = _parse_file(args.reppath, '%s.rpt' % rep_type, results,
-                              _extract_messages, rep_type, args)
-
-    # get gate equivalents
-    results = _parse_file(args.reppath, 'gate_equiv.rpt', results,
-                          _extract_gate_equiv, "area", args)
-    # area
-    results = _parse_file(args.reppath, 'area.rpt', results, _extract_area,
-                          "area", args)
-    # clocks. this complements the timing report later below
-    results = _parse_file(args.reppath, 'clocks.rpt', results, _extract_clocks,
-                          "timing", args)
-    # timing
-    results = _parse_file(args.reppath, 'timing.rpt', results, _extract_timing,
-                          "timing", args)
-    # power
-    results = _parse_file(args.reppath, 'power.rpt', results, _extract_power,
-                          "power", args)
-    # units
-    results = _parse_file(args.reppath, 'power.rpt', results, _extract_units,
-                          "units", args)
+    args.expand_modules = args.expand_modules.strip().split(',')
+
+    rep_types = [
+        (args.log_path, 'synthesis.log',  'flow',    _extract_messages),
+        (args.rep_path, 'analyze.rpt',    'analyze', _extract_messages),
+        (args.rep_path, 'elab.rpt',       'elab',    _extract_messages),
+        (args.rep_path, 'compile.rpt',    'compile', _extract_messages),
+        (args.rep_path, 'gate_equiv.rpt', 'area',    _extract_gate_equiv),
+        (args.rep_path, 'area.rpt',       'area',    _extract_area),
+        (args.rep_path, 'clocks.rpt',     'timing',  _extract_clocks),
+        (args.rep_path, 'timing.rpt',     'timing',  _extract_timing),
+        (args.rep_path, 'power.rpt',      'power',   _extract_power),
+        (args.rep_path, 'power.rpt',      'units',   _extract_units)
+    ]
+
+    for path, name, key, handler in rep_types:
+        _parse_file(path,
+                    name,
+                    results,
+                    handler,
+                    key,
+                    args)
```

Comment:
This is much nicer than what was there before! You can probably fit the `_parse_file` call on one line, though. Maybe a bit neater?

Reply:
jep, done!